src/include/sof/audio/selector.h

```diff
@@ -75,7 +70,7 @@ typedef void (*sel_func)(struct comp_dev *dev, struct audio_stream __sparse_cach
 /** \brief Selector component private data. */
 struct comp_data {
 #if CONFIG_IPC_MAJOR_4
-	struct micsel_data md;
+	struct ipc4_audio_format output_format;
```

Comment:
all ipc4 module is started with struct ipc4_base_module_cfg so it not good to remove the common header

Reply:
@RanderWang no, not all. eq-fir doesn't, nor does mfcc. So, instead of relying on every module to follow a rule, that we keep forgetting about, I put that configuration in a common place where it's available to all modules

Reply:
eq-fir doesn't support IPC4 now and does mfcc support it now ? I can't find mfcc. 
All the IPC4 module are derived from ipc4_base_module_cfg in C++ version and now we convert it to C version so ipc4_base_module_cfg  is at the head of structure.

Reply:
Yes, it is true for all ipc4 modules, please stay with ipc4_base_module_cfg

Reply:
> eq-fir doesn't support IPC4 now and does mfcc support it now ? I can't find mfcc. All the IPC4 module are derived from ipc4_base_module_cfg in C++ version and now we convert it to C version so ipc4_base_module_cfg is at the head of structure.

`CONFIG_COMP_FIR=y` and `CONFIG_COMP_MFCC=y` in IPC4 builds

Reply:
> Yes, it is true for all ipc4 modules, please stay with ipc4_base_module_cfg

@abonislawski could you explain why, please? How do you propose to address the bug, fixed in https://github.com/thesofproject/sof/pull/6426/commits/584df3d85859a7950cc7ac8e62b7fe7bc5cc949f ? Why would forcing each module to comply with a requirement be better than simply implementing that requirement once centrally?

Reply:
@lyakh It is included but it is not used.  These component is for IPC3 and I don't find eq_fir or mfcc in document for IPC4 and there is no commit to add ipc4 support for eq_fir or mfcc.

IPC3 component has a different UUID compared to IPC4, and we can find following code to support IPC4
````c
#if CONFIG_IPC_MAJOR_3
static const struct comp_driver comp_selector;

/* 55a88ed5-3d18-46ca-88f1-0ee6eae9930f */
DECLARE_SOF_RT_UUID("selector", selector_uuid, 0x55a88ed5, 0x3d18, 0x46ca,
		    0x88, 0xf1, 0x0e, 0xe6, 0xea, 0xe9, 0x93, 0x0f);
#else
/* 32fe92c1-1e17-4fc2-9758-c7f3542e980a */
DECLARE_SOF_RT_UUID("selector", selector_uuid, 0x32fe92c1, 0x1e17, 0x4fc2,
		    0x97, 0x58, 0xc7, 0xf3, 0x54, 0x2e, 0x98, 0x0a);

#endif
````

Reply:
@RanderWang how about https://github.com/thesofproject/sof/blob/ab3ad0151a9697a3aa36ad4e1a7be1583b3f0e9c/zephyr/wrapper.c#L538 ? That makes eq_fir not only build in IPC4 configurations but also link into the final image. If those components aren't available to IPC4 we have to stop building and linking them.

Reply:
Zephyr is not limited to IPC4.
Im not sure if you are aware but you have moved this for one or two modules when all the rest still have it in the original place so now we have in some modules one way and in some another way, this is just misleading rather than centrally fixing. Is this intentional?

Reply:
@abonislawski I removed those additional copies where I found them because they aren't needed any more. If I missed some, we can remove then in follow up PRs. Which modules have I missed?